src/main/java/net/consensys/orion/enclave/EncryptedPayload.java

```diff
@@ -90,19 +92,20 @@ public EncryptedPayload(
     return privacyGroupId;
   }
 
-  public EncryptedPayload stripFor(final Box.PublicKey key) {
-    final Integer toKeepIdx = encryptedKeyOwners.get(key);
+  public EncryptedPayload stripFor(final List<Box.PublicKey> keys) {
+    final List<EncryptedKey> keepKeys = new ArrayList<>();
+    keys.stream().filter(key -> encryptedKeys[encryptedKeyOwners.get(key)] != null).forEach(
+        key -> keepKeys.add(encryptedKeys[encryptedKeyOwners.get(key)]));
```

Comment:
Can you just use a map instead of the forEach and List.add

Reply:
Lot better! Done!